### ISSUE TITLE
Serializing Tasks (ie, not runnning in parallel)

### ISSUE THREAD

**Issue by asg017** [asg017/oak#79]
Say you have this Oakfile:
```

a
   \
    c
   / 
b
```

Right now, if both `a` and `b` are not fresh, they will most likely be ran at the same time. 

This is great in many cases, where you only care about the tasks being done fast, but in some cases, this can be problematic. If `a` and `b` are both resource heavy (RAM, gpu, etc.), then unexpected errors may occur.

So, let's have a way where users could explicitly define which tasks they want to run in serial. Adding an option to make everything serial may be impossible (given the runtime), but maybe we can have something like:

```javascript
m1 = new Mutex()

a = new Task({
  target: "a.txt",
  run: a=> shell`expensive_operation.sh "a" ${a}`,
  mutex: m1
})

b = new Task({
  target: "b.txt",
  run: b=> shell`expensive_operation.sh "b" ${b}`,
  mutex: m1
})

c = new Task({
  target: "c.txt",
  run: c => shell`cat ${a} ${b} > ${c}`
})
```

With this, we can explicitly define which tasks we want to be serialize with which tasks. So, either `a` or `b` will run first (the runtime can't exactly predict which will go first), and when it finishes, the next one will be in line to run. 

## What to call it?

I think mutex is the right term for what I'm thinking of (if my operating systems class taught me anything). Maybe "queue" or "lock" may make more sense (since many tasks can be queued up waiting for the lock to go to them).
